crates/flux-syntax/src/parser/mod.rs

```diff
@@ -93,7 +99,23 @@ fn parse_flux_item(cx: &mut ParseCtxt) -> ParseResult<Item> {
     }
 }
 
+fn advance_if_many(cx: &mut ParseCtxt, toks: &[crate::lexer::Token]) -> ParseResult<bool> {
+    for tok in toks {
+        if !cx.advance_if(*tok) {
+            return Ok(false);
+        }
+    }
+    Ok(true)
+}
+
+fn advance_if_pragma(cx: &mut ParseCtxt, tok: crate::lexer::Token) -> ParseResult<bool> {
+    let toks =
+        [Tok::Pound, Tok::OpenDelim(Delimiter::Bracket), tok, Tok::CloseDelim(Delimiter::Bracket)];
+    advance_if_many(cx, &toks)
+}
+
 fn parse_reft_func(cx: &mut ParseCtxt) -> ParseResult<SpecFunc> {
+    let hide = advance_if_pragma(cx, Tok::Hide)?;
```

Comment:
Since we don't yet have other attributes, I'd keep this specialized

```suggestion
fn parse_hide_attr(cx: &mut ParseCtxt) -> ParseResult<bool> {
    if !cx.advance_if(Tok::Pound) {
        return Ok(false);
    }
    cx.expect(Tok::OpenDelim(Bracket))?;
    cx.expect("hide")?;
    cx.expect(Tok::CloseDelim(Bracket))?;
    return Ok(true)
}

fn parse_reft_func(cx: &mut ParseCtxt) -> ParseResult<SpecFunc> {
    let hide = parse_hide_attr(cx);
```